src/js/structs/List.js

```diff
@@ -66,6 +66,26 @@ module.exports = class List {
   }
 
   /**
+   * Return a new list with the items currently present in this list and
+   * the list on the first argument, without duplicates.
+   *
+   * @param {List} list - The list to combine with this list
+   * @returns {List} Returns a new list with the combined items
+   */
+  combine(list) {
+    let myItems = this.getItems();
+    let newItems = list.reduceItems(function (items, item) {
+      if (myItems.indexOf(item) === -1) {
```

Comment:
Well, as a matter of fact `O(N²)` for such a core function, especially when it's used by the filters it's a bad idea. I will actually suggest a much faster yet a bit more complicate to read solution with `O(2N)` time.